src/components/KTableView/KTableView.vue

```diff
@@ -382,206 +382,55 @@ import type {
-    default: EmptyStateIconVariants.Default,
-  },
-  emptyStateButtonAppearance: {
-    type: String as PropType<ButtonAppearance>,
-    default: 'primary',
-  },
-  /**
-   * A prop that enables the error state
-   */
-  error: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-   * A prop to pass in a custom error state title
-   */
-  errorStateTitle: {
-    type: String,
-    default: 'An error occurred',
-  },
-  /**
-   * A prop to pass in a custom error state message
-   */
-  errorStateMessage: {
-    type: String,
-    default: 'Data cannot be displayed due to an error.',
-  },
-  /**
-   * A prop to pass in a custom error state action route
-   */
-  errorStateActionRoute: {
-    type: [Object, String] as PropType<RouteLocationRaw | string>,
-    default: null,
-  },
-  /**
-   * A prop to pass in a custom error state action message
-   */
-  errorStateActionMessage: {
-    type: String,
-    default: '',
-  },
-  /**
-   * A prop to pass in an array of headers for the table
-   */
-  headers: {
-    type: Array as PropType<TableViewHeader[]>,
-    default: () => [],
-  },
-  data: {
-    type: Array as PropType<TableViewData>,
-    default: () => [],
-  },
-  /**
-   * Provide the name of the data property keyto utilize as a unique identifier, or a function that receives the `row` object as a parameter that generates a unique identifier string for each row.
-   */
-  rowKey: {
-    type: [String, Function] as PropType<string | ((row: Record<string, any>) => string)>,
-    default: '',
-  },
-  maxHeight: {
-    type: String,
-    default: 'none',
-  },
-  hidePagination: {
-    type: Boolean,
-    default: false,
-  },
-  paginationAttributes: {
-    type: Object as PropType<TablePaginationAttributes>,
-    default: () => ({}),
-  },
-  /**
-   * Enable expandable rows
-   */
-  rowExpandable: {
-    type: Function as PropType<(row: Record<string, any>) => boolean>,
-    default: () => false,
-  },
-  /**
-   * Hide the table header
-   */
-  hideHeaders: {
-    type: Boolean,
-    default: false,
-  },
+import { getInitialPageSize } from '@/utilities'
+
+const props = withDefaults(defineProps<TableViewProps>(), {
+  resizeColumns: false,
+  tablePreferences: () => ({}),
+  rowHover: true,
+  rowAttrs: () => ({}),
+  rowLink: () => ({} as RowLink),
+  rowBulkActionEnabled: () => true,
+  rowKey: '',
+  cellAttrs: () => ({}),
+  loading: false,
+  emptyStateTitle: 'No Data',
+  emptyStateMessage: 'There is no data to display.',
+  emptyStateActionMessage: '',
+  emptyStateIconVariant: EmptyStateIconVariants.Default,
+  emptyStateButtonAppearance: 'primary',
+  error: false,
+  errorStateTitle: 'An error occurred',
+  errorStateMessage: 'Data cannot be displayed due to an error.',
+  errorStateActionMessage :'',
+  maxHeight: 'none',
+  hidePagination: false,
+  paginationAttributes: () => ({}),
+  rowExpandable: () => false,
+  hideHeaders: false,
+  nested: false,
+  hidePaginationWhenOptional: true,
```

Comment:
I think it's a bit confusing that `hidePaginationWhenOptional` defaults to `true` in KTableView, but defaults to `false` in `KTableData`

Reply:
Yes I realize this is rather misleading. Since KTableData implementation should ideally align with KTable (where it's `false` by default), I suggest that `hidePaginationWhenOptional` defaults to `false` in both KTableView and KTableData.

Reply:
Pushed a commit to change `hidePaginationWhenOptional` default value to `false` in KTableView.